### ISSUE TITLE
Menus in drawer do not close when clicking on the drawer

### ISSUE THREAD

**Issue by neclepsio** [material-components/material-components-web#579]
### What MDC-Web Version are you using?
0.10

### What browser(s) is this bug affecting?
I use Chrome, but it maybe affects others.

### What OS are you using?
Windows 7.

### What are the steps to reproduce the bug?
Have a menu inside a temporary drawer. 

### What is the expected behavior?
When the menu is open and click outside the menu but in the drawer, the menu should close.

### What is the actual behavior?
The menu stays open.

### Any other information you believe would be useful?
I think the problem may be that both drawer and menu use document click listener to close.

**Comment by mhuebert** [comment]
Probably caused by issue #225 - TemporaryDrawer is calling stopPropagation on all events that propagate upward from child elements.

**Comment by amsheehan** [comment]
This is the fourth issue we've seen regarding `stopPropagation` so this is clearly an issue. We will investigate ways to improve this behavior and update the community with our findings here. Thanks!

**Comment by whodidthis** [comment]
There's also something going on with drawer that makes [turbolinks](https://github.com/turbolinks/turbolinks) not work. 😱 

If you use new MDCPersistentDrawer or MDCTemporarydrawer on your drawer none of the links work via turbolinks anymore.

**Comment by ju1ius** [comment]
Hi,
Using `stopPropagation` also breaks event delegation on drawer child elements.
AFAICT, in this component `stopPropagation` is only used to prevent the drawer from closing when clicking inside the `.mdc-temporary-drawer__drawer` element.
One way to avoid this could be to use a real element for the drawer's backdrop instead of a pseudo-element.

```html
<aside class="mdc-temporary-drawer mdc-theme--dark">
  <div class="mdc-temporary-drawer__backdrop"></div>
  <nav class="mdc-temporary-drawer__drawer">
    ...
  </nav>
</aside>
```

**Comment by jamesmfriedman** [comment]
I found a solution to the issue but I wanted to get feedback before I submit a PR. Basically, React (and other libs) use event delegation and the drawer click handler on this line https://github.com/material-components/material-components-web/blob/65a5936a3e4ec82b09fd620d2053f761536916d1/packages/mdc-drawer/slidable/foundation.js#L57 prevents events from ever reaching the root node. StopProp can be evil.

My proposed solution is instead to put some close logic inside of the component clickHandler https://github.com/material-components/material-components-web/blob/65a5936a3e4ec82b09fd620d2053f761536916d1/packages/mdc-drawer/temporary/foundation.js#L45 that ignores closing if the sidebar was clicked, but still lets the events bubble.

Here is a simplified example...
```javascript
this.componentClickHandler_ = (evt) => {
  const path = evt.composedPath || evt.path || evt.deepPath;
  const drawerClickedWasClicked = path.some(el => {
    return el.classList && el.classList.contains('mdc-temporary-drawer__drawer')
  });

  if (!drawerClickedWasClicked) {
    this.close();
  }
}
```

One issue with this solution is that evt.path is newly standardized (Chrome calls it .path, and the standards track is composedPath). It can easily be polyfilled though.

Any thoughts on this solution?

**Comment by amsheehan** [comment]
Closing this. Please resume conversation on #1619

**Comment by kfranqueiro** [comment]
This has been fixed via 3e173e1ddc3bff5c7c9c2d794eeb36a9354ec98c (based on #1138), which refactors out the `stopPropagation` call.
